tests/test_ops/test_focal_loss.py

```diff
@@ -122,20 +117,66 @@ def _test_grad_sigmoid(self, dtype=torch.float):
             else:
                 gradcheck(floss, (x, y), eps=1e-2, atol=1e-2)
 
-    def test_softmax_float(self):
-        self._test_softmax(dtype=torch.float)
-
-    def test_softmax_half(self):
-        self._test_softmax(dtype=torch.half)
-
-    def test_sigmoid_float(self):
-        self._test_sigmoid(dtype=torch.float)
-
-    def test_sigmoid_half(self):
-        self._test_sigmoid(dtype=torch.half)
-
-    def test_grad_softmax_float(self):
-        self._test_grad_softmax(dtype=torch.float)
-
-    def test_grad_sigmoid_float(self):
-        self._test_grad_sigmoid(dtype=torch.float)
+    @pytest.mark.parametrize('device', [
+        'cpu',
+        pytest.param(
+            'cuda',
+            marks=pytest.mark.skipif(
+                not torch.cuda.is_available(),
+                reason='requires CUDA support')),
+    ])
+    def test_softmax_float(self, device):
+        self._test_softmax(torch.float, device)
+
+    @pytest.mark.parametrize('device', [
+        'cpu',
```

Comment:
We should also check whether CPU and CUDA output similar results when GPU is available.